open-banking-accelerator/components/event-notifications/com.wso2.openbanking.accelerator.event.notifications.service/src/main/java/com/wso2/openbanking/accelerator/event/notifications/service/handler/DefaultEventSubscriptionServiceHandler.java

```diff
@@ -168,22 +177,24 @@ public EventSubscriptionResponse getEventSubscriptionsByEventType(String clientI
             for (EventSubscription eventSubscription : eventSubscriptionList) {
                 eventSubscriptionResponseList.add(mapSubscriptionModelToResponseJson(eventSubscription));
             }
-            eventSubscriptionResponse.setStatus(EventNotificationConstants.OK);
+            eventSubscriptionResponse.setStatus(EventNotificationConstants.OK_CODE);
             eventSubscriptionResponse.setResponseBody(eventSubscriptionResponseList);
             return eventSubscriptionResponse;
         } catch (OBEventNotificationException e) {
             log.error("Error occurred while retrieving event subscriptions", e);
-            eventSubscriptionResponse.setStatus(EventNotificationConstants.INTERNAL_SERVER_ERROR);
-            eventSubscriptionResponse.setErrorResponse("Error occurred while retrieving event subscription.");
+            eventSubscriptionResponse.setStatus(EventNotificationConstants.INTERNAL_SERVER_ERROR_CODE);
+            eventSubscriptionResponse.setErrorResponse(EventNotificationServiceUtil.getErrorDTO(
+                    EventNotificationConstants.INVALID_REQUEST,
+                    "Error occurred while retrieving event subscription."));
```

Comment:
since we are catching a OBEventNotification exception, Can't we use the error messege passed through that? Please check.

Reply:
Fixed in https://github.com/wso2/financial-open-banking/pull/3/commits/a31eef6176e5003bd359e183aae45a90a5874f2b